### PR TITLE
add route optimization API docs

### PR DESCRIPTION
**Pull request by palmakkar**
## What?

add route optimization docs

### CODE REVIEW

File: docs/api.mdx [hunk 1]
```diff
@@ -1277,6 +1277,155 @@ curl "https://api.radar.io/v1/route/directions?locations=40.734250,-73.990927|40
 }
 ```
 
+#### Optimize
```
Comment (jsani-radar) [inline docs/api.mdx:1280]: the snap to road API is called "Route match" above - should this be "Route optimization" instead?

File: docs/api.mdx [hunk 2]
```diff
@@ -1277,6 +1277,155 @@ curl "https://api.radar.io/v1/route/directions?locations=40.734250,-73.990927|40
...
+
+Calculates the optimal ordering and route to visit a list of locations.
+
+###### Definitions
+
+<span className="badge badge--success">GET</span> <code>https://api.radar.io/v1/route/optimize</code>
+
+###### Query parameters
+
+- **`locations`** (string, required): List of up to 10 coordinates to visit. A pipe-delimited string in the format `latitude0,longitude0|latitude1,longitude1|...`. The first location should be the origin of the route, and the last location should be the final destination.
```
Comment (jsani-radar) [inline docs/api.mdx:1294]: nit: "A list" - see the Matrix API definition

File: docs/api.mdx [hunk 3]
```diff
@@ -1277,6 +1277,155 @@ curl "https://api.radar.io/v1/route/directions?locations=40.734250,-73.990927|40
...
+          "endLocation": {
+            "latitude": 40.7227423,
+            "longitude": -73.9983234
+          },
+          "startIndex": 0,
+          "endIndex": 3,
+          "geometry": {
+            "polyline": "kefulAxyiclCg_AlxCiBzFmBjGa}@`uC_AxCsAhEeBvEhJrEd^fQj@XfClAvAr@lHlD~BlAfFfClCpAtLjG~Av@vBfAp[lOpI`EjB|@tBbAdF`CtIbE`JfEjD`BhBz@lCnAlEtB|L|FnB~@xCvA|SzJ~E~BfK~EvJtE`Ab@~@d@fFbCjNvGl]lPlKbFxCvAdBx@bPvHhD`B~At@tPdIdCjAlBbAr\\lQdNvFhDvAvBXjEj@d[dChJt@dBNhCN~CNl_@zCjCTXsGzJc{BZgHmC{@oOwEmEsFaDaB}@eAe@yA\\gDvA_FvHsR~@aC~@cCjH}QzH_Sv@sBjAuChRef@`AeC|@}BhP}a@rAmDtAqD~Tkk@jTkj@`AcC|@_CzG}PxHwRbAkCnAaDjNg^bBqHBG~@mFfCrB~LzJvq@zi@xAjAbAx@fM`Kpw@xn@|AnAjAmDFShJ_Z"
+          },
+          "distance": {
```
Comment (jsani-radar) [inline docs/api.mdx:1345]: nit: should we pull the `distance` and `duration` structs higher in the payload? `polyline` is pretty noisy

File: docs/api.mdx [hunk 4]
```diff
@@ -1277,6 +1277,155 @@ curl "https://api.radar.io/v1/route/directions?locations=40.734250,-73.990927|40
...
+            "value": 55616.79968,
+            "text": "10.5 mi"
+          },
+          "duration": {
+            "value": 15.5131,
+            "text": "16 mins"
+          }
+        },
```
Comment (jsani-radar) [inline docs/api.mdx:1422]: nit: don't think the comma is needed if it's the final closing bracket

**Review by jsani-radar — COMMENTED** [review]
small comments, looking good overall

File: docs/api.mdx (hunk 1 above)
Comment (palmakkar) [reply docs/api.mdx:1280]: yea thanks for catching, got it mixed up with the explore page pattern

**Review by jsani-radar — APPROVED** [review]

**Pull request merged**